### DISCUSSION TITLE
Add `merged_branch_labels` or similar to allow for removing branch labels from later history

### DISCUSSION THREAD

**Discussion by kageurufu** [sqlalchemy/alembic#1022] [Ideas]
We use alembic branches to track features as we build them in parallel to main development, and have ended up with many superfluous labels on our history. The only way I've found to "resolve" a branch label is to remove it from the original migration, which has it's own issues. 

I propose adding a property such as `merge_branch_labels` to migrations, which would remove the label from future revisions. This is mostly a cosmetic change, but would be very welcome

**Comment by zzzeek** [comment]
im not following what you mean by "resolve" can you provide a concrete example?

also it seems like the proposal is a change to the versioning model such that there would be some way that a descendant of a certain label would somehow no longer be considered as part of that branch label, which seems like it would be quite awkward.

**Comment by kageurufu** [comment]
It's possible I'm looking at branching completely wrong in Alembic, but here goes.
We often use git branching for large feature development, and when the branch is merged, the history is kept but the branch no longer exists.

Here's an example history for how I would see it in use (integer revisions and spacing added for readability)

```
4               -> 5 (head),                             Continued work after accounting is complete
3, accounting_3 -> 4 (mergepoint) (resolves accounting), Merge accounting branch
2               -> 3,                                    mainline development
accounting_2    -> accounting_3 (accounting),            last revision for accounting
accounting_1    -> accounting_2 (accounting),            Add accounting platform
2               -> accounting_1 (accounting),            Add accounting platform
1               -> 2 (branchpoint),                      Add some tables
<base>          -> 1,                                    Initial
```

After the mergepoint at `4` is made, making a migration on an alias like `accounting@head` would no longer be possible, and the branch would no longer be shown on later revisions.

The annoyance is in long-living projects, you might end up with dozens of branch labels on a single revision. Its probable in retrospect that we should have used `depends_on` for all this and kept many branches around?

Here is an example alembic setup to reproduce what we've done [example.tar.gz](https://github.com/sqlalchemy/alembic/files/8557839/example.tar.gz)

**Reply by zzzeek** [reply]
OK, the proposal seems like I understood what you meant, but the use case is different.   it seems that seeing all the branch labels in....`alembic history`, is unsightly?  What's wrong with the dozens of branch labels?

**Reply by zzzeek** [reply]
like, if this is a visual adjustment to the commands, that's much more doable
